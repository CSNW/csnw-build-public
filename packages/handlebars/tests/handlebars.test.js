const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

const handlebars = require('../')({
  partials: './src/partials'
});
const cwd = join(__dirname, 'fixtures');

test('pipeline', async () => {
  const process = await handlebars.pipeline({
    cwd,
    sourcemaps: true,
    cache: false
  });
  const files = await Files.from('**/*', { cwd, baseDir: join(cwd, 'src') });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
