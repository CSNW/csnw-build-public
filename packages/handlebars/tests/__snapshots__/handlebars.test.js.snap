// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "templates.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/handlebars/tests/fixtures/src",
    "contents": "window[\\"TEMPLATES\\"] = window[\\"TEMPLATES\\"] || {};
window[\\"TEMPLATES\\"][\\"a\\"] = Handlebars.template({\\"compiler\\":[7,\\">= 4.0.0\\"],\\"main\\":function(container,depth0,helpers,partials,data) {
    var stack1;

  return \\"Hello \\"
    + ((stack1 = container.invokePartial(partials.name,depth0,{\\"name\\":\\"name\\",\\"data\\":data,\\"helpers\\":helpers,\\"partials\\":partials,\\"decorators\\":container.decorators})) != null ? stack1 : \\"\\")
    + ((stack1 = container.invokePartial(partials.punctuation,depth0,{\\"name\\":\\"punctuation\\",\\"data\\":data,\\"helpers\\":helpers,\\"partials\\":partials,\\"decorators\\":container.decorators})) != null ? stack1 : \\"\\");
},\\"usePartial\\":true,\\"useData\\":true});
Handlebars.registerPartial('name', Handlebars.template({\\"compiler\\":[7,\\">= 4.0.0\\"],\\"main\\":function(container,depth0,helpers,partials,data) {
    return \\"NAME\\";
},\\"useData\\":true}));
Handlebars.registerPartial('punctuation', Handlebars.template({\\"compiler\\":[7,\\">= 4.0.0\\"],\\"main\\":function(container,depth0,helpers,partials,data) {
    return \\"!\\";
},\\"useData\\":true}));",
    "cwd": "C:/dev/csnw-build/packages/handlebars/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/handlebars/tests/fixtures/src/templates.js",
    "sourceMap": null,
  },
}
`;
